# Projeto: Women Empowerment Index
Este notebook realiza a análise exploratória do dataset Women Empowerment Index.
Objetivos:
- Entender estrutura e qualidade dos dados
- Limpar inconsistências
- Extrair insights iniciais


In [6]:
# Importar bibliotecas
import pandas as pd

## Estrutura inicial dos dados
Verificando dimensões, colunas e tipos para entender a base.

In [7]:
# Leitura dos dados

df = pd.read_csv("women_empowerment_index.csv")

In [10]:
df.shape
df.info("women_empowerment_index.csv")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114 entries, 0 to 113
Data columns (total 7 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   Country                                   114 non-null    object 
 1   Women's Empowerment Index (WEI) - 2022    114 non-null    float64
 2   Women's Empowerment Group - 2022          114 non-null    object 
 3   Global Gender Parity Index (GGPI) - 2022  114 non-null    float64
 4   Gender Parity Group - 2022                114 non-null    object 
 5   Human Development Group - 2021            114 non-null    object 
 6   Sustainable Development Goal regions      114 non-null    object 
dtypes: float64(2), object(5)
memory usage: 6.4+ KB


In [11]:
# 1) Renomear colunas para nomes mais curtos e sem espaços
df = df.rename(columns={
    "Women's Empowerment Index (WEI) - 2022": "WEI_2022",
    "Women's Empowerment Group - 2022": "WE_Group_2022",
    "Global Gender Parity Index (GGPI) - 2022": "GGPI_2022",
    "Gender Parity Group - 2022": "Gender_Parity_Group_2022",
    "Human Development Group - 2021": "HD_Group_2021",
    "Sustainable Development Goal regions": "SDG_Region"
})

# 2) Checar duplicados
print("Linhas duplicadas:", df.duplicated().sum())

# 3) Ver valores únicos por coluna categórica
for col in ['Country','WE_Group_2022','Gender_Parity_Group_2022','HD_Group_2021','SDG_Region']:
    print(f"\nColuna: {col}")
    print(df[col].unique())


Linhas duplicadas: 0

Coluna: Country
['Australia' 'Belgium' 'Denmark' 'Iceland' 'Norway' 'Sweden' 'Austria'
 'Bulgaria' 'Canada' 'Croatia' 'Czechia' 'Estonia' 'Finland' 'France'
 'Germany' 'Hungary' 'Ireland' 'Italy' 'Latvia' 'Lithuania' 'Luxembourg'
 'Netherlands' 'Poland' 'Portugal' 'Serbia' 'Singapore' 'Slovenia' 'Spain'
 'Switzerland' 'United Kingdom' 'United States' 'Albania' 'Armenia'
 'Bolivia (Plurinational State of)' 'Brazil' 'Chile' 'China' 'Costa Rica'
 'Cyprus' 'Dominican Republic' 'Ecuador' 'Greece' 'Israel' 'Jamaica'
 'Japan' 'Malta' 'Mauritius' 'Moldova (Republic of)' 'Mongolia'
 'Montenegro' 'Namibia' 'North Macedonia' 'Peru' 'Philippines' 'Romania'
 'Slovakia' 'South Africa' 'Thailand' 'United Arab Emirates' 'Uruguay'
 'Viet Nam' 'Bangladesh' 'Belize' 'Benin' 'Bhutan'
 'Bosnia and Herzegovina' 'Botswana' 'Burkina Faso' 'Burundi' 'Cambodia'
 'Cameroon' 'Colombia' 'Congo (Democratic Republic of the)'
 "Côte d'Ivoire" 'Egypt' 'El Salvador' 'Gambia' 'Ghana' 'Guatemala'
 '

## Limpeza e padronização
- Renomeamos as colunas para nomes curtos (ex.: `WEI_2022`) para facilitar queries e gráficos.
- Verificamos duplicados e analisamos valores únicos das colunas categóricas.


In [13]:
df = df.rename(columns={
    "Women's Empowerment Index (WEI) - 2022": "WEI_2022",
    "Women's Empowerment Group - 2022": "WE_Group_2022",
    "Global Gender Parity Index (GGPI) - 2022": "GGPI_2022",
    "Gender Parity Group - 2022": "Gender_Parity_Group_2022",
    "Human Development Group - 2021": "HD_Group_2021",
    "Sustainable Development Goal regions": "SDG_Region"
})

print("Linhas duplicadas:", df.duplicated().sum())


Linhas duplicadas: 0


# 📊 Análise Exploratória – Women Empowerment Index 2022

Nesta etapa vamos explorar o conjunto de dados já limpo para:
- Entender a distribuição dos índices numéricos (WEI e GGPI).
- Identificar países e regiões com maiores e menores valores.
- Investigar padrões por grupo de desenvolvimento humano e região.
- Avaliar a relação entre o Empowerment Index (WEI) e o Gender Parity Index (GGPI).

---

## 1️⃣ Estatísticas Descritivas

Objetivo: verificar médias, medianas, mínimos e máximos para ter uma visão geral da variação dos índices.


In [14]:
df[['WEI_2022','GGPI_2022']].describe()

WEI_2022   GGPI_2022
count  114.000000  114.000000
mean     0.607316    0.705237
std      0.134632    0.128897
min      0.141000    0.141000
25%      0.518000    0.622250
50%      0.615000    0.726500
75%      0.705500    0.802000
max      0.828000    0.925000

## 2️⃣ Ranking dos Países

Objetivo: identificar os 10 países com **maiores** e **menores** índices de empoderamento feminino (WEI).


In [15]:
# Top 10
df.nlargest(10, 'WEI_2022')[['Country','WEI_2022']]

# Bottom 10
df.nsmallest(10, 'WEI_2022')[['Country','WEI_2022']]


Country  WEI_2022
111                               Yemen     0.141
97                                Niger     0.307
99                             Pakistan     0.337
84                                 Iraq     0.363
88                              Lebanon     0.372
72   Congo (Democratic Republic of the)     0.399
79                               Guinea     0.400
93                                 Mali     0.400
63                                Benin     0.418
76                               Gambia     0.427

## 3️⃣ Comparações por Grupos e Regiões

Objetivo: descobrir quais regiões e grupos de desenvolvimento humano apresentam as maiores médias do índice.


In [16]:
# Média por região (Sustainable Development Goal)
df.groupby('SDG_Region')['WEI_2022'].mean().sort_values(ascending=False)

# Média por grupo de Desenvolvimento Humano
df.groupby('HD_Group_2021')['WEI_2022'].mean().sort_values(ascending=False)


HD_Group_2021
Very high    0.726261
High         0.595808
Medium       0.528864
Low          0.435000
Name: WEI_2022, dtype: float64

## 4️⃣ Correlação entre WEI e GGPI

Objetivo: analisar a relação entre o Women Empowerment Index e o Global Gender Parity Index.


In [17]:
df[['WEI_2022','GGPI_2022']].corr()


WEI_2022  GGPI_2022
WEI_2022   1.000000   0.839409
GGPI_2022  0.839409   1.000000

## 5️⃣ Visualizações Rápidas

Objetivo: criar gráficos simples para observar padrões e dispersões.


## 🌍 Distribuição Global do Women Empowerment Index – 2022

O mapa abaixo apresenta a variação do **Women Empowerment Index (WEI)** em 114 países.  
Quanto mais escura a cor, maior o índice de empoderamento feminino em 2022.


In [21]:
import plotly.express as px

fig = px.choropleth(
    df,
    locations="Country",
    locationmode="country names",
    color="WEI_2022",
    color_continuous_scale="YlGnBu",
    title="Distribuição Global do Women Empowerment Index (2022)"
)
fig.show()

### 📝 Insights Principais
- **Liderança**: [0.828] Países da Europa Ocidental e América do Norte exibem índices acima de **0.80**, indicando políticas consolidadas de igualdade de gênero.  
- **Baixos índices**: [0.141] Regiões da África Subsaariana permanecem abaixo de **0.40**, evidenciando maiores desafios sociais e econômicos.  
- **Tendência global**: A diferença entre as regiões mais e menos desenvolvidas revela oportunidades de cooperação internacional e investimento em programas de empoderamento feminino.
